desktop/libs/azure/src/azure/abfs/upload.py

```diff
@@ -227,3 +233,54 @@ def _get_scheme(self):
         raise ABFSFileSystemException('Destination does not start with a valid scheme.')
     else:
       return None
+
+
+class ABFSNewFileUploadHandler(ABFSFileUploadHandler):
+  """
+  This handler uploads the file to ABFS if the destination path starts with "ABFS" (case insensitive).
+  Streams data chunks directly to ABFS.
+  """
+
+  def __init__(self, dest_path, username):
+    self.chunk_size = DEFAULT_WRITE_SIZE
+    self.target_path = None
+    self.file = None
+    self._part_size = DEFAULT_WRITE_SIZE
+
+    self.destination = dest_path
+    self.username = username
+
+    # TODO: _is_abfs_upload really required?
+    if self._is_abfs_upload():
+      self._fs = self._get_abfs(self.username)
+      self.filesystem, self.directory = parse_uri(self.destination)[:2]
+
+      # Verify that the path exists
+      self._fs.stats(self.destination)
```

Comment:
like in ofs, add exception here as well

Reply:
Relooking at the code, I removed all these destination exists check from all upload handlers because we are doing this check at filebrowser API level so it was redundant doing it again in all upload handlers.